File: src/services/apiAdapt/openapp.ts
import { dual, dualVoid } from './shared';

export interface AppStatus {
  installed: boolean;
  path: string | null;
}

export async function checkAppInstalled(appName: string) {
  return await dual<AppStatus>(
    'check_app_installed',
    { appName },
    '/api/openapp/check-app-installed',
    { appName }
  );
}

export async function openWorkspaceIn(
  path: string,
  options: {
    appName?: string | null;
    command?: string | null;
    args?: string[];
  }
) {
  const payload = {
    app: options.appName ?? null,
    command: options.command ?? null,
    args: options.args ?? [],
  };
  await dualVoid(
    'open_workspace_in',
    { path, options: payload },
    '/api/openapp/open-workspace-in',
    { path, options: payload }
  );
}

File: src-tauri/src/commands/openapp.rs
use codexia_shared::openapp::{self, AppStatus, OpenWorkspaceOptions};

#[tauri::command]
pub async fn check_app_installed(app_name: String) -> Result<AppStatus, String> {
    openapp::check_app_installed(app_name).await
}

#[tauri::command]
pub async fn open_workspace_in(path: String, options: OpenWorkspaceOptions) -> Result<(), String> {
    openapp::open_workspace_in(path, options).await
}
